res/controllers/Denon-MC7000-scripts.js

```diff
@@ -486,7 +507,15 @@ MC7000.PadButtons = function(channel, control, value, status, group) {
             midi.sendShortMsg(0x94 + deckNumber - 1, 0x14 + i, MC7000.padColor.rolloff);
         }
     } else if (MC7000.PADModeSavedLoop[deckNumber]) {
-        return;
+        if (value === 0x00) {
+            return; // don't respond to note off messages
+        }
+        i = control - 0x14;
```

Comment:
I don't see `i` declared in any outer scope so this defaults to a global variable.
```suggestion
        var i = control - 0x14;
```
Also same issue in line 501

Reply:
it is already defined in line 483

Reply:
It isn't. That var is inside an if case. When this code gets executed, the code in the if-case won't. the variable here is still not declared and thus JS defaults to making it global. Having a global variable named `i` will likely cause bugs in the future.

Reply:
but it is inside one function. If I write `var i` in each of the `if` / `else if` statements then the pre-commit check will not pass as it says the variables are defined already and multiple definitions are not allowed.

Reply:
Interesting, I guess thats because the legacy `var` keyword is function scoped (instead of block-scoped as the more modern `let` and `const` are). I still don't like the way it is currently, but it doesn't have the implicit-global problem I was concerned about.